manifests/jail.pp

```diff
@@ -14,6 +14,7 @@
   Integer $bantime = $fail2ban::bantime,
   Optional[String] $port = undef,
   Optional[String] $backend = undef,
+  Optional[Array] $ignoreip = undef,
```

Comment:
Please take a look at our [review guidelines](https://voxpupuli.org/docs/#reviewing-a-module-pr):

> Are there new params with datatype Hash or Array? If possible, they should default to empty Hash/Array instead of undef. You can also enforce the datastructure like Array[String]

Can you please migration the Datatype to something like `Array[String[1]]`? What kind of ip addresses are valid? Stdlib has [quite some datatypes](https://github.com/puppetlabs/puppetlabs-stdlib#stdlibipaddress). One of them is probably useful in this case so it could even be `Array[Stdlib::IP::Address]`

Reply:
We probably need to update the lowest supported stdlib version in the metadata.json as well, since the new types got introduces in stdlib [4.25.0](https://github.com/puppetlabs/puppetlabs-stdlib/commit/6dad21e292436ac5b7d8113e81e572a0efc66782).